staging/operator-lifecycle-manager/pkg/controller/operators/decorators/operator.go

```diff
@@ -127,14 +127,32 @@ func (o *Operator) ComponentLabelKey() (string, error) {
 		return o.componentLabelKey, nil
 	}
 
-	if o.GetName() == "" {
+	name := o.GetName()
+	if name == "" {
 		return "", fmt.Errorf(componentLabelKeyError, "empty name field")
 	}
 
-	name := o.GetName()
 	if len(name) > 63 {
 		// Truncate
 		name = name[0:63]
+
+		// Remove trailing illegal characters
+		idx := len(name) - 1
+		for ; idx >= 0; idx-- {
+			lastChar := name[idx]
+			if lastChar != '.' && lastChar != '_' && lastChar != '-' {
+				break
+			}
+		}
+
+		// Just being defensive. This is unlikely to happen since the operator name should
+		// be compatible kubernetes naming constraints
+		if idx < 0 {
+			return "", fmt.Errorf(componentLabelKeyError, "unsupported name field")
+		}
+
+		// Update Label
+		name = name[0 : idx+1]
```

Comment:
Not sure why no one seems to:
```
if name = strings.TrimRight(name[0:63], "._-"); len(name) <= 0 {
    return ....
}
```
Should the illegal characters be checked outside the `len(name)` check? Is it possible to set a name with an illegal trailing character? Could that happen?

Reply:
> Not sure why no one seems to:
> 
> if name = strings.TrimRight(name[0:63], "._-"); len(name) <= 0 {
>     return ....
> }
I actually prefer this as well, but this is a backport of a fix and we attempt to avoid altering the code. We can always create a PR on master to do this though.

> Is it possible to set a name with an illegal trailing character? Could that happen?

The kube api server will reject the create request and OLM will constantly try to create it even with the error, which is why we're backporting this.